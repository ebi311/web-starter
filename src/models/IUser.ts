export type Gender = 'man' | 'woman' | 'none';
export default interface IUser {
  familyName: string;
  givenName: string;
  gender: Gender;
}
