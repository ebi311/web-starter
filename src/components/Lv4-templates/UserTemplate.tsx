import React from 'react';
import { styled } from '../Theme';
import IUser from '../../models/IUser';
import UserForm from '../Lv3-organisms/UserForm';

const Container = styled.div`
  border: 1px solid ${p => p.theme.foreground1};
  padding: 1em;
  margin: 1em;
`;

interface IProps {
  user: IUser;
}

const UserTemplate: React.FC<IProps> = props => {
  return (
    <Container>
      <UserForm user={props.user} />
    </Container>
  );
};

export default UserTemplate;
