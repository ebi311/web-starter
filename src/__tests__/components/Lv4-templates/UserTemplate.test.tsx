import React from 'react';
import { shallow } from 'enzyme';
import UserTemplate from '../../../components/Lv4-templates/UserTemplate';
import IUser from '../../../models/IUser';

describe('UserTemplate', () => {
  const user: IUser = {
    familyName: 'nanashi',
    givenName: 'gonbe',
    gender: 'man',
  };
  test('success', () => {
    const wrapper = shallow(<UserTemplate user={user} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
