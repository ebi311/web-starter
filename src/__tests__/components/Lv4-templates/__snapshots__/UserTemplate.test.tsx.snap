// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTemplate success 1`] = `
<div
  className="sc-AxmLO jYPplo"
>
  <UserForm
    user={
      Object {
        "familyName": "nanashi",
        "gender": "man",
        "givenName": "gonbe",
      }
    }
  />
</div>
`;
